### ISSUE TITLE
About MPC Solver Method ： SLQ and SQP

### ISSUE THREAD

**Issue by DiomandLEE** [utiasDSL/upright#7]
Hello, sorry to bother you again. Do you know what practical problems the OCS2 SQP (multiple shooting) and SLQ (DDP) are suitable for? I see that only the quadruped robot example uses the SQP option, while the other examples use the SLQ algorithm of DDP. Also, the useFeedbackPolicy option is set to false for SLQ and true for SQP. 
<img src="https://github.com/utiasDSL/upright/assets/87216511/ce429751-d836-44f4-8b9a-c6b506321b71"   height="400" />
<img src="https://github.com/utiasDSL/upright/assets/87216511/6dbcb5f0-6efb-48f4-b56f-1193a4f6d42c" width="400" height="300" />

I'm not sure why this is the case, but I noticed that you are using the SQP algorithm. What was your reason for choosing this? Additionally, you changed SensitivityIntegratorType::RK2 to SensitivityIntegratorType::RK4. Why is that?

**Comment by adamheins** [comment]
DDP does not have a way to handle hard inequality constraints, which means they need to be added to the cost as penalties. For this work, I found this led to weird cost landscapes and odd behaviour that was difficult to tune, which is why I switched to SQP. I also actually found SQP to run a bit faster.

Regarding changing the integrator, I don't have a really strong reason for doing this. While doing some initial tuning I think I changed that setting and it didn't make a big difference to performance either way, if I recall correctly.
